writer.go

```diff
@@ -278,6 +289,13 @@ func (p *MediaPlaylist) last() uint {
 // Remove current segment from the head of chunk slice form a media playlist. Useful for sliding playlists.
 // This operation does reset playlist cache.
 func (p *MediaPlaylist) Remove() (err error) {
+	p.lock.Lock()
+	err = p.remove()
+	p.lock.Unlock()
```

Comment:
For code consistency, please p.lock.Lock() followed by defer p.lock.Unlock()

Reply:
fixed